File: frontend/src/types/shop.ts
export interface Shop {
  id: number;
  name: string;
  category: string | null;
  category_name: string | null;
  icon: string | null;
  latitude: number;
  longitude: number;
  address: string | null;
  phone: string | null;
  image_url: string | null;
  rating: number;
  review_count: number;
  is_open: boolean;
  closing_time: string | null;
  distance?: number;
}

export interface ShopCreatePayload {
  name: string;
  category_slug?: string;
  latitude: number;
  longitude: number;
  address?: string;
}

File: backend/app/schemas/shop.py
from pydantic import BaseModel


class ShopCreate(BaseModel):
    name: str
    category_slug: str | None = None
    latitude: float
    longitude: float
    address: str | None = None
    phone: str | None = None
    image_url: str | None = None


class ShopUpdate(BaseModel):
    name: str | None = None
    category_slug: str | None = None
    latitude: float | None = None
    longitude: float | None = None
    address: str | None = None
    phone: str | None = None
    image_url: str | None = None
    is_open: bool | None = None
    closing_time: str | None = None


class ShopOut(BaseModel):
    id: int
    name: str
    category: str | None = None
    category_name: str | None = None
    icon: str | None = None
    latitude: float
    longitude: float
    address: str | None = None
    phone: str | None = None
    image_url: str | None = None
    rating: float
    review_count: int
    is_open: bool
    closing_time: str | None = None
    distance: float | None = None

    model_config = {"from_attributes": True}
